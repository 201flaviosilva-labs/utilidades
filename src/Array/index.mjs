import { allEqual } from "./allEqual.mjs";
import { choice } from "./choice.mjs";
import { findBigObject } from "./findBigObject.mjs";
import { findLowObject } from "./findLowObject.mjs";
import { isSorted } from "./isSorted.mjs";
import { moveLeft, moveRight } from "./Move.mjs";
import { shuffle } from "./shuffle.mjs";
import { sortAscending } from "./sortAscending.mjs";
import { sortAscendingObject } from "./sortAscendingObject.mjs";
import { sortDescending } from "./sortDescending.mjs";
import { sortDescendingObject } from "./sortDescendingObject.mjs";

/**
 * Functions utils for arrays
 * 
 * @namespace Array
 */
export {
	allEqual,
	choice,
	findBigObject,
	findLowObject,
	isSorted,
	moveLeft, moveRight,
	shuffle,
	sortAscending, sortAscendingObject,
	sortDescending, sortDescendingObject,
};
