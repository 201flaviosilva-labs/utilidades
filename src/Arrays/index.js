import { allEqual } from "./allEqual.js";
import { choice } from "./choice.js";
import { findBigObject } from "./findBigObject.js";
import { findLowObject } from "./findLowObject.js";
import { isSorted } from "./isSorted.js";
import { moveLeft, moveRight } from "./move.js";
import { shuffle } from "./shuffle.js";
import { sortAscending } from "./sortAscending.js";
import { sortAscendingObject } from "./sortAscendingObject.js";
import { sortDescending } from "./sortDescending.js";
import { sortDescendingObject } from "./sortDescendingObject.js";

/**
 * Functions utils for arrays
 * 
 * @namespace Arrays
 */
export {
	allEqual,
	choice,
	findBigObject,
	findLowObject,
	isSorted,
	moveLeft, moveRight,
	shuffle,
	sortAscending, sortAscendingObject,
	sortDescending, sortDescendingObject,
};
