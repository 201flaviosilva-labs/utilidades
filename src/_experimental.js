// Points
export function distanceTwoPoints(x1, y1, x2, y2) { return Math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)); }
export function distanceTwoPointsVector2(point1, point2) { return distanceTwoPoints(point1.x, point1.y, point2.x, point2.y); }

export function angleBetweenTwoPoints(x1, y1, x2, y2) { return Math.atan2(y2 - y1, x2 - x1); }
export function angleBetweenTwoPointsVector2(point1, point2) { return angleBetweenTwoPoints(point1.x, point1.y, point2.x, point2.y); }
export function angleBetweenTwoPointsDegrees(x1, y1, x2, y2) { return angleBetweenTwoPointsRadians(x1, y1, x2, y2) * 180 / Math.PI; }
export function angleBetweenTwoPointsVector2Degrees(point1, point2) { return angleBetweenTwoPointsDegrees(point1.x, point1.y, point2.x, point2.y); }

// Circles
export function distanceBetweenTwoCirclesFromCircle(circle1, circle2) {
	return distanceTwoPoints(circle1.x, circle1.y, circle2.x, circle2.y) - (circle1.radius + circle2.radius);
}
export function distanceBetweenTwoCircles(x1, y1, radius1, x2, y2, radius2) {
	return distanceBetweenTwoCirclesFromCircle(
		{ x: x1, y: y1, radius: radius1 },
		{ x: x2, y: y2, radius: radius2 }
	);
}

export function circleArea(radius) { return Math.PI * radius * radius; }
export function circlePerimeter(radius) { return 2 * Math.PI * radius; }

// Rectangles
export function distanceBetweenTwoRectanglesFromBounds(rect1, rect2) {
	const xOverlap = Math.min(rect1.x + rect1.width, rect2.x + rect2.width) - Math.max(rect1.x, rect2.x);
	const yOverlap = Math.min(rect1.y + rect1.height, rect2.y + rect2.height) - Math.max(rect1.y, rect2.y);
	return xOverlap * yOverlap;
}
export function distanceBetweenTwoRectangles(x1, y1, width1, height1, x2, y2, width2, height2) {
	return distanceBetweenTwoRectanglesFromBounds(
		{ x: x1 + width1 / 2, y: y1 + height1 / 2, width: width1, height: height1 },
		{ x: x2 + width2 / 2, y: y2 + height2 / 2, width: width2, height: height2 }
	);
}

export function angleBetweenTwoRectanglesFromBounds(rect1, rect2) {
	const xOverlap = Math.min(rect1.x + rect1.width, rect2.x + rect2.width) - Math.max(rect1.x, rect2.x);
	const yOverlap = Math.min(rect1.y + rect1.height, rect2.y + rect2.height) - Math.max(rect1.y, rect2.y);
	return Math.atan2(yOverlap, xOverlap);
}
export function angleBetweenTwoRectangles(x1, y1, width1, height1, x2, y2, width2, height2) {
	return angleBetweenTwoRectanglesFromBounds(
		{ x: x1 + width1 / 2, y: y1 + height1 / 2, width: width1, height: height1 },
		{ x: x2 + width2 / 2, y: y2 + height2 / 2, width: width2, height: height2 }
	);
}

export function rectangleArea(width, height) { return width * height; }
export function rectanglePerimeter(width, height) { return 2 * (width + height); }
export function rectanglePerimeterFromBounds(rectangle) { return rectanglePerimeter(rectangle.x, rectangle.y, rectangle.width, rectangle.height); }

export function rectangleVertices(x, y, width, height) {
	return [
		{ x: x, y: y },
		{ x: x + width, y: y },
		{ x: x + width, y: y + height },
		{ x: x, y: y + height }
	];
}
export function rectangleVerticesFromBounds(rectangle) { return rectangleVertices(rectangle.x, rectangle.y, rectangle.width, rectangle.height); }

export function rectangleCenterX(x, width) { return x + width / 2; }
export function rectangleCenterY(y, height) { return y + height / 2; }
export function rectangleCenter(x, y, width, height) { return { x: rectangleCenterX(x, width), y: rectangleCenterY(y, height) }; }
export function rectangleCenterXFromBounds(rectangle) { return rectangleCenterX(rectangle.x, rectangle.width); }
export function rectangleCenterYFromBounds(rectangle) { return rectangleCenterY(rectangle.y, rectangle.height); }
export function rectangleCenterFromBounds(rectangle) { return rectangleCenter(rectangle.x, rectangle.y, rectangle.width, rectangle.height); }
