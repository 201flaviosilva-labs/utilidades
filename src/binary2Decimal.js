/**
 * Convert a binary number to a decimal
 * 
 * @example
 * binary2Decimal("101010"); // 42
 * 
 * @param {string} binary - the value to be converted
 * @returns {number} - conversion decimal
 */
export function binary2Decimal(binary) {
	if (typeof binary === "string") return parseInt(binary, 2);
}
