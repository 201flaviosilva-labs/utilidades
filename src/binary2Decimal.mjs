/**
 * Convert a number/string to a decimal
 * 
 * @example
 * binary2Decimal(101010); // 42
 * 
 * @param {number|string} binary - the number/string to be converted
 * @returns {number} - conversion decimal
 */
export function binary2Decimal(binary) {
	if (typeof binary === "string") return parseInt(binary.split("").reverse().join(""), 2);
	else if (typeof binary === "number") return parseInt(binary, 2);
}
