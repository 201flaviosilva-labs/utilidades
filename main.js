import * as utilidades from "./src/main";

console.log(utilidades);

document.querySelector("#app").innerHTML = `
  <div>
    <a href="https://vitejs.dev" target="_blank">
      <img src="/vite.svg" class="logo" alt="Vite logo" />
    </a>
    <h1>Utilidades Testes</h1>
  </div>
`;
