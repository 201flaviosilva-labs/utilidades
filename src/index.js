import * as Arrays from "./Arrays/index.js";
import * as Maths from "./Maths/index.js";
import * as Physics from "./Physics/index.js";

import { allCharactersSame } from "./allCharactersSame.js";
import { binary2Decimal } from "./binary2Decimal.js";
import { BinarySearchTree, BinarySearchTreeInstance } from "./BinarySearchTree.js";
import { choiceTrend } from "./choiceTrend.js";
import { clone } from "./clone.js";
import { compare2Objects } from "./compare2Objects.js";
import { decimal2Binary } from "./decimal2Binary.js";
import { getDate } from "./getDate.js"; // getTime, getMilliseconds, getSeconds, getMinutes, getHours, getDay, getWeekDay, getWeek, getMonth, getYear, getDateFormatted 
import { EventSystem, EventSystemInstance } from "./EventSystem.js";
import { Fibonacci, fibonacciSequence, fibonacciUntil, fibonacciCustomSequence, recursiveFibonacci } from "./Fibonacci.js";
import { getVersion } from "./getVersion.js";
import { isObjectEmpty } from "./isObjectEmpty.js";
import { isFalsy, isTruthy } from "./isTruthyFalsy.js";
import { and, or, xor } from "./logicalOperators.js";
import { randomColor, randomColor0X, randomRGBColor, randomRGBAColor } from "./randomColor.js";
import { randomNumber, randomInt, randomFloat } from "./randomNumber.js";
import { randomString } from "./randomString.js";
import { randomWalk1D, randomWalk2D, randomWalk3D } from "./randomWalk.js";
import { reverseString } from "./reverseString.js";
import { Vector2 } from "./Vector2.js";

export {
	Arrays,
	Maths,
	Physics,

	BinarySearchTree, BinarySearchTreeInstance,
	EventSystem, EventSystemInstance,
	Fibonacci, fibonacciSequence, fibonacciUntil, fibonacciCustomSequence, recursiveFibonacci,
	Vector2,

	allCharactersSame,
	binary2Decimal,
	compare2Objects,
	decimal2Binary,
	choiceTrend,
	clone,
	getDate,
	getVersion,
	isObjectEmpty,
	isFalsy, isTruthy,
	and, or, xor, // logicalOperators
	randomColor, randomColor0X, randomRGBColor, randomRGBAColor,
	randomNumber, randomInt, randomFloat,
	randomString,
	randomWalk1D, randomWalk2D, randomWalk3D,
	reverseString,
};
