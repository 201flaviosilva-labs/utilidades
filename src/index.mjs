import * as Array from "./Array/index.mjs";
// import * as DOM from "./DOM/index.mjs";
import * as Maths from "./Maths/index.mjs";
// import * as SortingAlgorithms from "./SortingAlgorithms/index.mjs";

import { allCharactersSame } from "./allCharactersSame.mjs";
import { binary2Decimal } from "./binary2Decimal.mjs";
import { BinarySearchTree, BinarySearchTreeInstance } from "./BinarySearchTree.mjs";
import { stringToCamelCase, camelCaseToNormal } from "./camelCase.mjs";
import { stringToCapitalize, capitalizeToNormal } from "./capitalizeCase.mjs";
import { choiceTrend } from "./choiceTrend.mjs";
import { clone } from "./clone.mjs";
import { compare2Objects } from "./compare2Objects.mjs";
import { decimal2Binary } from "./decimal2Binary.mjs";
import { getDate } from "./getDate.mjs"; // getTime, getMilliseconds, getSeconds, getMinutes, getHours, getDay, getWeekDay, getWeek, getMonth, getYear, getDateFormatted 
import { EventSystem, EventSystemInstance } from "./EventSystem.mjs";
import { Fibonacci, fibonacciSequence, fibonacciUntil, fibonacciCustomSequence, recursiveFibonacci } from "./Fibonacci.mjs";
import { FIFO } from "./FIFO.mjs";
// import { getUrlParameter } from "./getUrlParameter.mjs";
// import { getVersion } from "./getVersion.mjs";
import { invertSentence, invertWords } from "./invertText.mjs";
import { isFalsy, isTruthy } from "./isTruthyFalsy.mjs";
import { stringToKebabCase, kebabCaseToNormal } from "./kebabCase.mjs";
import { and, or, xor } from "./logicalOperators.mjs";
import { LIFO } from "./LIFO.mjs";
import { randomColor, randomColor0X, randomRGBColor, randomRGBAColor } from "./randomColor.mjs";
import { randomNumber, randomInt, randomFloat } from "./randomNumber.mjs";
import { randomString } from "./randomString.mjs";
import { randomWalk1D, randomWalk2D, randomWalk3D } from "./randomWalk.mjs";
import { stringToScreamingSnakeCase, screamingSnakeCaseToNormal } from "./screamingSnakeCase.mjs";
import { stringToSnakeCase, snakeCaseToNormal } from "./snakeCase.mjs";
import { topDownCarMovimentation } from "./topDownCarMovimentation.mjs";
import { Vector2 } from "./Vector2.mjs";

export {
	Array,
	// DOM,
	Maths,
	// SortingAlgorithms,

	BinarySearchTree, BinarySearchTreeInstance,
	EventSystem, EventSystemInstance,
	Fibonacci, fibonacciSequence, fibonacciUntil, fibonacciCustomSequence, recursiveFibonacci,
	FIFO,
	LIFO,
	Vector2,

	allCharactersSame,
	binary2Decimal,
	stringToCamelCase, camelCaseToNormal,
	stringToCapitalize, capitalizeToNormal,
	choiceTrend,
	clone,
	compare2Objects,
	decimal2Binary,
	getDate,
	// getUrlParameter,
	// getVersion,
	invertSentence, invertWords,
	isFalsy, isTruthy,
	stringToKebabCase, kebabCaseToNormal,
	and, or, xor, // logicalOperators
	randomColor, randomColor0X, randomRGBColor, randomRGBAColor,
	randomNumber, randomInt, randomFloat,
	randomString,
	randomWalk1D, randomWalk2D, randomWalk3D,
	stringToScreamingSnakeCase, screamingSnakeCaseToNormal,
	stringToSnakeCase, snakeCaseToNormal,
	topDownCarMovimentation,
};
