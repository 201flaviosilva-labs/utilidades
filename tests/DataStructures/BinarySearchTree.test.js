import { describe, expect, it, test } from "vitest";
import { DataStructures } from "./src/main";
const { BinarySearchTree, BinarySearchTreeInstance, BSTNode: Node } = DataStructures;

describe("BinarySearchTree.js", () => {
	/**
	 * 						100
	 * 					/		\
	 * 				50			125
	 *		          \			/   \
	 * 				  75	  111	150		
	 * 				 /  \
	 * 				66   88
	 */
	const values = [100, 50, 125, 111, 75, 66, 150, 88];

	describe("BinarySearchTree", () => {
		describe("constructor()", () => {
			it("should the root be null", () => {
				const bst = new BinarySearchTree();

				expect(bst.root).toBe(null);
				expect(bst.count).toBe(0);
				expect(bst).toMatchSnapshot();
			});

			it("should the root node have the given value", () => {
				const bst = new BinarySearchTree(100);

				expect(bst.root).toBeInstanceOf(Node);
				expect(bst.root.value).toBe(100);
				expect(bst.root.left).toBe(null);
				expect(bst.root.right).toBe(null);

				expect(bst.count).toBe(1);
			});

			it("should the root node have the given array [100, 50, 125]", () => {
				const bst = new BinarySearchTree([100, 50, 125]);

				expect(bst.root).toBeInstanceOf(Node);
				expect(bst.root.value).toBe(100);
				expect(bst.root.left.value).toBe(50);
				expect(bst.root.right.value).toBe(125);

				expect(bst.count).toBe(3);
			});

			it("should throw an error if the value is not a number", () => {
				expect(() => new BinarySearchTree("A")).toThrow("Node value must be a number.");
			});

			it("should match a full tree with the snapshot", () => {
				const bst = new BinarySearchTree(values);

				expect(bst).toMatchSnapshot();
			});
		});

		describe("add()", () => {
			it("should add a single value to the left [10, 5]", () => {
				const bst = new BinarySearchTree(10);

				const root = bst.root;
				expect(root.value).toBe(10);

				bst.add(5);
				const n1 = root.left;
				expect(n1.value).toBe(5);
			});

			it("should add a single value to the right [10, 20]", () => {
				const bst = new BinarySearchTree(10);

				const root = bst.root;
				expect(root.value).toBe(10);

				bst.add(20);
				const n1 = root.right;
				expect(n1.value).toBe(20);
			});

			it("should add a single value to the left and right [10, 5, 20]", () => {
				const bst = new BinarySearchTree(10);

				const root = bst.root;
				expect(root.value).toBe(10);

				bst.add(5);
				const l1 = root.left;
				expect(l1.value).toBe(5);

				bst.add(20);
				const r1 = root.right;
				expect(r1.value).toBe(20);
			});

			it("should not add a value that already exists", () => {
				const bst = new BinarySearchTree(0);

				bst.add(0);
				expect(bst.root.value).toBe(0);
				expect(bst.root.left).toBe(null);
				expect(bst.root.right).toBe(null);
			});

			test("[E2E] should add multiples mixed values: [100, 50, 125, 111, 75, 66, 150, 88]", () => {
				const bst = new BinarySearchTree();

				bst.add(100);
				bst.add(50); // left
				bst.add(125); // right
				bst.add(111); // right -> left
				bst.add(75); // left -> right
				bst.add(66); // left -> right -> left
				bst.add(150); // right -> right
				bst.add(88); // left -> right -> right

				const root = bst.root;
				expect(root.value).toBe(100);

				const left = root.left;
				expect(left.value).toBe(50);

				const right = root.right;
				expect(right.value).toBe(125);

				const rightLeft = right.left;
				expect(rightLeft.value).toBe(111);

				const leftRight = left.right;
				expect(leftRight.value).toBe(75);

				const leftRightLeft = leftRight.left;
				expect(leftRightLeft.value).toBe(66);

				const rightRight = right.right;
				expect(rightRight.value).toBe(150);

				const leftRightRight = leftRight.right;
				expect(leftRightRight.value).toBe(88);
			});
		});

		describe("smaller()", () => {
			it("should return the smallest value in the tree", () => {
				const bst = new BinarySearchTree(values);
				expect(bst.smaller()).toBe(50);
			});

			it("should return the smallest value in the tree from a given value", () => {
				const bst = new BinarySearchTree(values);
				expect(bst.smaller(100)).toBe(50);
				expect(bst.smaller(50)).toBe(50);
				expect(bst.smaller(125)).toBe(111);
				expect(bst.smaller(111)).toBe(111);
				expect(bst.smaller(75)).toBe(66);
				expect(bst.smaller(66)).toBe(66);
				expect(bst.smaller(150)).toBe(150);
				expect(bst.smaller(88)).toBe(88);
			});

			it("should return undefined if the tree is empty", () => {
				const bst = new BinarySearchTree();
				expect(bst.smaller()).toBeUndefined();
			});

			it("should return null if the value is not in the tree", () => {
				const bst = new BinarySearchTree(values);

				expect(bst.smaller(0)).toBeNull();
				expect(bst.smaller(-100)).toBeNull();
				expect(bst.smaller(55)).toBeNull();
				expect(bst.smaller(1000)).toBeNull();
			});
		});

		describe("larger()", () => {
			it("should return the largest value in the tree", () => {
				const bst = new BinarySearchTree(values);
				expect(bst.larger()).toBe(150);
			});

			it("should return the largest value in the tree from a given value", () => {
				const bst = new BinarySearchTree(values);
				expect(bst.larger(100)).toBe(150);
				expect(bst.larger(50)).toBe(88);
				expect(bst.larger(125)).toBe(150);
				expect(bst.larger(111)).toBe(111);
				expect(bst.larger(75)).toBe(88);
				expect(bst.larger(66)).toBe(66);
				expect(bst.larger(150)).toBe(150);
				expect(bst.larger(88)).toBe(88);
			});

			it("should return undefined if the tree is empty", () => {
				const bst = new BinarySearchTree();
				expect(bst.larger()).toBeUndefined();
			});

			it("should return null if the value is not in the tree", () => {
				const bst = new BinarySearchTree(values);

				expect(bst.larger(0)).toBeNull();
				expect(bst.larger(-100)).toBeNull();
				expect(bst.larger(55)).toBeNull();
				expect(bst.larger(1000)).toBeNull();
			})
		});

		describe("search()", () => {
			it("should return the node/value", () => {
				const bst = new BinarySearchTree(values);

				expect(bst.search(100)).toBeInstanceOf(Node);

				expect(bst.search(100).value).toBe(100);
				expect(bst.search(50).value).toBe(50);
				expect(bst.search(125).value).toBe(125);
				expect(bst.search(150).value).toBe(150);
				expect(bst.search(88).value).toBe(88);

				expect(bst.search(100, false)).toBe(true);
				expect(bst.search(50, false)).toBe(true);
				expect(bst.search(125, false)).toBe(true);
			});

			it("should return undefined if the tree is empty", () => {
				const bst = new BinarySearchTree();
				expect(bst.search()).toBeUndefined();
			});

			it("should return null if the value is not in the tree", () => {
				const bst = new BinarySearchTree(values);

				expect(bst.search(0)).toBeNull();
				expect(bst.search(-100)).toBeNull();
				expect(bst.search(55)).toBeNull();
				expect(bst.search(1000)).toBeNull();
			});

			it("should return a error if the value is not a number", () => {
				const bst = new BinarySearchTree(values);

				expect(() => bst.search("A")).toThrowError("Value need to be a number!");
			});
		});

		describe("delete()", () => {
			it("should remove the root node", () => {
				const bst = new BinarySearchTree(100);

				bst.delete(100);
				expect(bst.search(100)).toBeUndefined();
				expect(bst.count).toBe(0);
			});

			it("should remove an no child node", () => {
				const bst = new BinarySearchTree([100, 50, 150]);

				// Left
				bst.delete(50);
				expect(bst.search(50)).toBeNull();
				expect(bst.count).toBe(2);

				// Right
				bst.delete(150);
				expect(bst.search(150)).toBeNull();
				expect(bst.count).toBe(1);
			});

			it("should remove the root node with one left child", () => {
				const bst = new BinarySearchTree([100, 50]);

				bst.delete(100);
				expect(bst.search(100)).toBeNull();
				expect(bst.root.value).toBe(50);
				expect(bst.count).toBe(1);
			});

			it("should remove an one child node on the left", () => {
				const bst = new BinarySearchTree([100, 50, 25]);

				bst.delete(50);
				expect(bst.search(50)).toBeNull();

				expect(bst.root.left.value).toBe(25);
				expect(bst.count).toBe(2);
			});

			it("should remove an one child node on the right", () => {
				const bst = new BinarySearchTree([100, 125, 150]);

				bst.delete(125);
				expect(bst.search(125)).toBeNull();

				expect(bst.root.right.value).toBe(150);
				expect(bst.count).toBe(2);
			});

			it("should remove a node with two children", () => {
				const bst = new BinarySearchTree(values);

				bst.delete(125);
				expect(bst.search(125)).toBeNull();

				expect(bst.root.right.value).toBe(111);
				expect(bst.root.right.left).toBeNull();
				expect(bst.root.right.right.value).toBe(150);
				expect(bst.count).toBe(7);
			});

			it("should return null if node is not found", () => {
				const bst = new BinarySearchTree(values);
				expect(bst.delete(0, true)).toBeNull();
			});
		})
	});

	describe("Node", () => {
		it("should have a value", () => {
			const node = new Node(100);

			expect(node.value).toBe(100);
			expect(node.left).toBeNull();
			expect(node.right).toBeNull();
			expect(node).toMatchSnapshot();
		});

		it("should return a error if the value is not a number", () => {
			expect(() => new Node()).toThrowError("Node value cannot be undefined.");
			expect(() => new Node(false)).toThrowError("Node value must be a number.");
			expect(() => new Node("A")).toThrowError("Node value must be a number.");
		});
	});
});
