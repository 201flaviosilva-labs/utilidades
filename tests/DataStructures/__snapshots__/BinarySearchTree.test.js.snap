// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BinarySearchTree.js > BinarySearchTree > constructor() > should match a full tree with the snapshot 1`] = `
BinarySearchTree {
  "count": 8,
  "root": Node {
    "left": Node {
      "left": null,
      "right": Node {
        "left": Node {
          "left": null,
          "right": null,
          "value": 66,
        },
        "right": Node {
          "left": null,
          "right": null,
          "value": 88,
        },
        "value": 75,
      },
      "value": 50,
    },
    "right": Node {
      "left": Node {
        "left": null,
        "right": null,
        "value": 111,
      },
      "right": Node {
        "left": null,
        "right": null,
        "value": 150,
      },
      "value": 125,
    },
    "value": 100,
  },
}
`;

exports[`BinarySearchTree.js > BinarySearchTree > constructor() > should the root be null 1`] = `
BinarySearchTree {
  "count": 0,
  "root": null,
}
`;

exports[`BinarySearchTree.js > Node > should have a value 1`] = `
Node {
  "left": null,
  "right": null,
  "value": 100,
}
`;
